src/sdam/description/server.rs

```diff
@@ -106,7 +107,36 @@ pub(crate) struct ServerDescription {
     // allows us to ensure that only valid states are possible (e.g. preventing that both an error
     // and a reply are present) while still making it easy to define helper methods on
     // ServerDescription for information we need from the hello reply by propagating with `?`.
-    pub(crate) reply: Result<Option<HelloReply>, String>,
+    pub(crate) reply: Result<Option<HelloReply>>,
```

Comment:
The fact that we store `String` here instead of the actual error is kind of a historical quirk at this point. In order to implement the full error matching for the test runner, I needed to make this a regular `Result`. IMO this is an improvement anyways.

Reply:
at least for the workload executor it doesn't really matter what the format is. all of the serialized stuff just gets dumped into JSON logs